packages/transaction-controller/src/TransactionController.ts

```diff
@@ -578,17 +578,18 @@ export class TransactionController extends BaseController<
         addInitialHistorySnapshot(transactionMeta);
       }
 
+      transactions.push(transactionMeta);
+      this.update({
+        transactions: this.trimTransactionsForState(transactions),
+      });
+
```

Comment:
Below, in `updateSimulationData`, update operation happens for `simulationData`, it obtains `TransactionMeta` from state of the controller. However, `transactionMeta` doesn't exist in the case of `addTransaction` method. So moving up these update methods will fix that issue.